dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassPathEntry.java

```diff
@@ -112,13 +113,18 @@ private void readClassFileNames() throws IOException {
       this.classFileNames = classNames.build();
     }
   }
-  
+
   /**
-   * Returns the fully qualified names of the classes in this entry's jar file.
-   * 
+   * Returns a list of class file names in {@link #jar} as in {@link JavaClass#getFileName()}. This
+   * class file name is usually a fully qualified class name. However a class file name may have a
+   * framework-specific prefix. Example: {@code BOOT-INF.classes.com.google.Foo}.
+   *
+   * <p>This function caches the result. Therefore the caller does not need to worry about the
```

Comment:
I'm hesitant to expose implementation details in API doc

Reply:
Removed the mention about the caching.